From 1f9384267efbf2f9679610db423f601f94b8522b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:14:15 -0400
Subject: [redhat] infra: initial ppcle changes

Message-id: <1410545655-205645-627-git-send-email-dzickus@redhat.com>
Patchwork-id: 94606
O-Subject: [RHEL7 PATCH 626/626] redhat: initial ppcle changes
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

Initial cut of getting the ppc64le to build.  Easily tested by building
a src.rpm and

brew build --scratch --arch-override=ppc64le rhel-7.1-ppc64le-auto <src.rpm>

Then installing the resulting rpm on a ppc64le guest and booting it to
verify it works.  And it does. :-)

Config options taken from the ppc64le project in Fedora.

No kabi added yet...
---
 create mode 100644 redhat/configs/generic/powerpc64le/CONFIG_BPF_JIT
 create mode 100644 redhat/configs/generic/powerpc64le/CONFIG_CARL9170
 create mode 100644 redhat/configs/generic/powerpc64le/CONFIG_CPU_LITTLE_ENDIAN
 create mode 100644 redhat/configs/generic/powerpc64le/CONFIG_VIRTUALIZATION

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/configs/build_configs.sh b/redhat/configs/build_configs.sh
index e84bc63..92e5f2b 100755
--- a/redhat/configs/build_configs.sh
+++ b/redhat/configs/build_configs.sh
@@ -40,6 +40,8 @@ function merge_configs()
  done
  if [ "x$arch" == "xppc64" ]; then
   echo "# powerpc" > $name
+ elif [ "x$arch" == "xppc64le" ]; then
+  echo "# powerpc" > $name
  elif [ "x$arch" == "xs390x" ]; then
   echo "# s390" > $name
  else
diff --git a/redhat/configs/generic/powerpc64le/CONFIG_BPF_JIT b/redhat/configs/generic/powerpc64le/CONFIG_BPF_JIT
new file mode 100644
index 0000000..7905dce
--- /dev/null
+++ b/redhat/configs/generic/powerpc64le/CONFIG_BPF_JIT
@@ -0,0 +1 @@
+# CONFIG_BPF_JIT is not set
diff --git a/redhat/configs/generic/powerpc64le/CONFIG_CARL9170 b/redhat/configs/generic/powerpc64le/CONFIG_CARL9170
new file mode 100644
index 0000000..cff12d5
--- /dev/null
+++ b/redhat/configs/generic/powerpc64le/CONFIG_CARL9170
@@ -0,0 +1 @@
+# CONFIG_CARL9170 is not set
diff --git a/redhat/configs/generic/powerpc64le/CONFIG_CPU_LITTLE_ENDIAN b/redhat/configs/generic/powerpc64le/CONFIG_CPU_LITTLE_ENDIAN
new file mode 100644
index 0000000..ee43fdb
--- /dev/null
+++ b/redhat/configs/generic/powerpc64le/CONFIG_CPU_LITTLE_ENDIAN
@@ -0,0 +1 @@
+CONFIG_CPU_LITTLE_ENDIAN=y
diff --git a/redhat/configs/generic/powerpc64le/CONFIG_VIRTUALIZATION b/redhat/configs/generic/powerpc64le/CONFIG_VIRTUALIZATION
new file mode 100644
index 0000000..da6788b
--- /dev/null
+++ b/redhat/configs/generic/powerpc64le/CONFIG_VIRTUALIZATION
@@ -0,0 +1 @@
+# CONFIG_VIRTUALIZATION is not set
diff --git a/redhat/configs/priority b/redhat/configs/priority
index 377bca8..e215ba6 100644
--- a/redhat/configs/priority
+++ b/redhat/configs/priority
@@ -9,6 +9,9 @@ x86_64-debug=generic:generic-x86_64:debug:debug-x86_64
 # ppc64
 ppc64=generic:generic-powerpc64
 ppc64-debug=generic:generic-powerpc64:debug:debug-powerpc64
+# ppc64le
+ppc64le=generic:generic-powerpc64:generic-powerpc64le
+ppc64le-debug=generic:generic-powerpc64:generic-powerpc64le:debug:debug-powerpc64
 # s390x
 s390x=generic:generic-s390x
 s390x-debug=generic:generic-s390x:debug:debug-s390x
diff --git a/redhat/git/files b/redhat/git/files
index 31632ad..5f1f16b 100644
--- a/redhat/git/files
+++ b/redhat/git/files
@@ -21,6 +21,8 @@ kabi_whitelist_s390x
 kabi_whitelist_x86_64
 kernel-3.10.0-ppc64.config
 kernel-3.10.0-ppc64-debug.config
+kernel-3.10.0-ppc64le.config
+kernel-3.10.0-ppc64le-debug.config
 kernel-3.10.0-s390x.config
 kernel-3.10.0-s390x-debug.config
 kernel-3.10.0-s390x-kdump.config
diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 346ac02..f442e04 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -104,12 +104,12 @@ Summary: The Linux kernel
 %endif
 
 # These arches install vdso/ directories.
-%define vdso_arches %{all_x86} x86_64 ppc ppc64 s390 s390x
+%define vdso_arches %{all_x86} x86_64 ppc ppc64 ppc64le s390 s390x
 
 # Overrides for generic default options
 
-# only build kernel-debug on x86_64, s390x, ppc64
-%ifnarch x86_64 s390x ppc64
+# only build kernel-debug on x86_64, s390x, ppc64 ppc64le
+%ifnarch x86_64 s390x ppc64 ppc64le
 %define with_debug 0
 %endif
 
@@ -129,7 +129,7 @@ Summary: The Linux kernel
 %endif
 
 # sparse blows up on ppc64
-%ifarch ppc64 ppc
+%ifarch ppc64 ppc64le ppc
 %define with_sparse 0
 %endif
 
@@ -152,7 +152,7 @@ Summary: The Linux kernel
 %define hdrarch powerpc
 %endif
 
-%ifarch ppc64
+%ifarch ppc64 ppc64le
 %define asmarch powerpc
 %define hdrarch powerpc
 %define all_arch_configs kernel-%{version}-ppc64*.config
@@ -205,7 +205,7 @@ Summary: The Linux kernel
 %endif
 
 # Architectures we build tools/cpupower on
-%define cpupowerarchs x86_64 ppc64
+%define cpupowerarchs x86_64 ppc64 ppc64le
 
 #
 # Three sets of minimum package version requirements in the form of Conflicts:
@@ -275,7 +275,7 @@ Version: %{rpmversion}
 Release: %{pkg_release}
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
-ExclusiveArch: noarch i686 x86_64 ppc ppc64 s390 s390x
+ExclusiveArch: noarch i686 x86_64 ppc ppc64 ppc64le s390 s390x
 ExclusiveOS: Linux
 
 %kernel_reqprovconf
@@ -355,6 +355,8 @@ Source51: kernel-%{version}-x86_64-debug.config
 
 Source60: kernel-%{version}-ppc64.config
 Source61: kernel-%{version}-ppc64-debug.config
+Source62: kernel-%{version}-ppc64le.config
+Source63: kernel-%{version}-ppc64le-debug.config
 
 Source70: kernel-%{version}-s390x.config
 Source71: kernel-%{version}-s390x-debug.config
@@ -901,7 +903,7 @@ BuildKernel() {
     fi
     rm -f $RPM_BUILD_ROOT/lib/modules/$KernelVer/build/scripts/*.o
     rm -f $RPM_BUILD_ROOT/lib/modules/$KernelVer/build/scripts/*/*.o
-%ifarch ppc64
+%ifarch ppc64 ppc64le
     cp -a --parents arch/powerpc/lib/crtsavres.[So] $RPM_BUILD_ROOT/lib/modules/$KernelVer/build/
 %endif
     if [ -d arch/%{asmarch}/include ]; then
-- 
1.7.1